presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -366,6 +372,112 @@ else if (firstDependencies.stream().allMatch(right::canResolve) && secondDepende
         return new RelationPlan(root, analysis.getScope(node), outputSymbols);
     }
 
+    private RelationPlan planJoinUsing(Join node, RelationPlan left, RelationPlan right)
+    {
+        /* Given: l JOIN r USING (k1, ..., kn)
```

Comment:
IMO this comment is not needed. There are plenty of places which requires knowledge about SQL semantic.

Reply:
It helps understand what's the overall structure of the resulting. Sometimes, it's hard to reverse-engineer from the manipulation of nodes and fields needed.